### ISSUE TITLE
React false positive in a pure Vue application

### ISSUE THREAD

**Issue by crystalfp** [oxc-project/oxc#3329]
False positive for a Pinia defined function. Note the application is pure Vue, so the React error is really misplaced.
```
 eslint-plugin-react-hooks(rules-of-hooks): React Hook "useConfigStore" cannot be called in a class component. React Hooks must be called in a React function component or a custom React Hook function.
     ╭─[src\nodes\Symmetries.ts:160:23]
 159 │         // Update the UI
 160 │         const configStore = useConfigStore();
     ·                             ────────────────
 161 │         configStore.control.computedSpaceGroup = outSymmetry;
     ╰────
```

**Comment by Boshen** [comment]
I think the best thing to do a here is omit the rule by `-A rules-of-hooks` 🤔

**Comment by Boshen** [comment]
Please also note that `nursery` rules are expected to raise false positives.

Close as won't fix because you can disable this rule in a vue application via `-A rules-of-hooks`.
